inspirehep/dojson/hep/fields/bd1xx.py

```diff
@@ -24,76 +24,137 @@
 
 
+ORCID = re.compile('\d{4}-\d{4}-\d{4}-\d{3}[0-9X]')
+
+
 @hep.over('authors', '^[17]00[103_].')
 def authors(self, key, value):
-    """Main Entry-Personal Name."""
-    value = force_force_list(value)
-
-    def get_value(value):
-        affiliations = []
-        if value.get('u'):
-            recid = None
-            try:
-                recid = int(value.get('z'))
-            except:
-                pass
-            affiliations = force_force_list(value.get('u'))
-            record = get_record_ref(recid, 'institutions')
-            affiliations = [{'value': aff, 'record': record} for
-                            aff in affiliations]
-        person_recid = None
-        if value.get('x'):
-            try:
-                person_recid = int(value.get('x'))
-            except:
+    """Authors.
+
+    FIXME: currently not handling 100__v.
+    """
+    def _get_author(value):
+        def _get_affiliations(value):
+            result = []
+
+            u_values = force_force_list(value.get('u'))
+            z_values = force_force_list(value.get('z'))
+
+            if len(u_values) == len(z_values):
+                for u_value, z_value in zip(u_values, z_values):
+                    result.append({
+                        'record': get_record_ref(z_value, 'institutions'),
+                        'value': u_value,
+                    })
+            elif len(u_values) > len(z_values):
+                # TODO: complain to @kaplun
+                for u_value, z_value in zip_longest(u_values, z_values):
+                    result.append({
+                        'record': get_record_ref(z_value, 'institutions'),
+                        'value': u_value,
+                    })
+            else:
+                # TODO: loudly complain to @kaplun
                 pass
-        inspire_id = ''
-        if value.get('i'):
-            inspire_id = force_force_list(value.get('i'))[0]
-        person_record = get_record_ref(person_recid, 'authors')
-        ret = {
-            'full_name': value.get('a'),
-            'role': value.get('e'),
+
+            return result
+
+        def _get_full_name(value):
+            a_values = force_force_list(value.get('a'))
+            if a_values:
+                if len(a_values) > 1:
+                    current_app.logger.error(
+                        'Record with mashed up authors list. '
+                        'Taking first author: %s', a_values[0]
+                    )
+
+                return a_values[0]
+
+        def _get_ids(value):
+            result = []
+
+            i_values = force_force_list(value.get('i'))
+            for i_value in i_values:
+                result.append({
+                    'type': 'INSPIRE ID',
+                    'value': i_value,
+                })
+
+            j_values = force_force_list(value.get('j'))
+            for j_value in j_values:
+                if j_value.startswith('JACoW-') or j_value.startswith('JACOW-'):
```

Comment:
The plan here is to have:
```python
def _is_jacow(j_value):
    return j_value.upper().startswith('JACOW-')
```
and similar helpers. A bit on the Lispy side, with functions nested 4 levels deep...